akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala

```diff
@@ -316,6 +316,23 @@ private[http] object StreamUtils {
     }
     Flow[T].transformMaterializing(newForeachStage)
   }
+
+  /**
+   * Similar to Source.lazyEmpty but doesn't rely on materialization. Can only be used once.
```

Comment:
I see, so it exposes the `OneTimeValce#open()` before materialization so you can complete it eagerly... Ok!